### PR TITLE
distutils-r1.eclass: Replace fancy inline Python with external gpep517 project

### PR DESCRIPTION
**Pull request by mgorny**
CC @gentoo/python

### CODE REVIEW

File: dev-python/gpep517/metadata.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "https://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<maintainer type="person">
+		<email>mgorny@gentoo.org</email>
+		<name>Michał Górny</name>
+	</maintainer>
+	<upstream>
```
Comment (arthurzam) [inline dev-python/gpep517/metadata.xml:8]: Should it be ALLARCHES?

File: eclass/distutils-r1.eclass [hunk 2]
```diff
@@ -1008,35 +996,22 @@ distutils_pep517_install() {
...
+			die "Wheel build failed"
 	)
 	[[ -n ${wheel} ]] || die "No wheel name returned"
 
 	einfo "  Installing the wheel to ${root}"
-	# NB: --compile-bytecode does not produce the correct paths,
-	# and python_optimize doesn't handle being called outside D,
-	# so we just defer compiling until the final merge
-	# NB: we override sys.prefix & sys.exec_prefix because otherwise
-	# installer would use virtualenv's prefix
```
Comment (arthurzam) [inline eclass/distutils-r1.eclass:1031]: Just a question: is this comment irrelevant with new nicer code?

File: dev-python/gpep517/metadata.xml (hunk 1 above)
Comment (mgorny) [reply dev-python/gpep517/metadata.xml:8]: Yes.

File: eclass/distutils-r1.eclass (hunk 2 above)
Comment (mgorny) [reply eclass/distutils-r1.eclass:1031]: The first one is still kinda relevant but I've pretty much assumed it's status quo to be doing compiling later. The second one is not.

**Comment by mgorny** [comment]
Ok, here's a better version that uses `GPEP517_TESTING` to switch between old and new code. This one can effectively be merged and tested without breaking stable tree ;-).

**Pull request closed without merging**